### PR TITLE
RAV-8571: yyyymmddhh24missms date format

### PR DESCRIPTION
**Pull request by akashshukla-dm**

### CODE REVIEW

**Comment by nosheenSheikh02** [comment]
The code looks fine but add some test cases for it.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -9511,4 +9516,15 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
         .withBigQuery()
         .ok(expectedBQSql);
   }
+
+  @Test public void testdateFormat() {
+    String query =
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:9521]: remove this

**Review by nosheenSheikh02 — APPROVED** [review]

File: core/src/main/java/org/apache/calcite/sql/SqlDateTimeFormat.java [hunk 2]
```diff
@@ -47,6 +47,9 @@
   TWENTYFOURHOURMIN("HH24MI"),
   TWENTYFOURHOURMINSEC("HH24MISS"),
   YYYYMMDDHH24MISS("YYYYMMDDHH24MISS"),
+  YYYYMMDDHH24MISSMS("YYYYMMDDHH24MISSMS"),
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/sql/SqlDateTimeFormat.java:50]: This are not required, these should be in Raven, please connect with someone from hive team for the logic.

**Comment by akashshukla-dm** [comment]
Hi @soma-mondal , i have made the required changes on Mig side and updated calcite changes accordingly. Please review now.

**Pull request merged**